crates/turborepo-lib/src/turbo_json/parser.rs

```diff
@@ -320,6 +321,100 @@ impl Deserializable for ConfigurationOptions {
     }
 }
 
+impl Deserializable for RawRemoteCacheOptions {
+    fn deserialize(
+        value: &impl DeserializableValue,
+        name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        value.deserialize(RawRemoteCacheOptionsVisitor, name, diagnostics)
+    }
+}
+
+struct RawRemoteCacheOptionsVisitor;
+
+impl DeserializationVisitor for RawRemoteCacheOptionsVisitor {
+    type Output = RawRemoteCacheOptions;
+
+    const EXPECTED_TYPE: VisitableType = VisitableType::MAP;
+
+    fn visit_map(
+        self,
+        // Iterator of key-value pairs.
+        members: impl Iterator<Item = Option<(impl DeserializableValue, impl DeserializableValue)>>,
+        // range of the map in the source text.
+        _: TextRange,
+        _name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self::Output> {
+        let mut result = RawRemoteCacheOptions::default();
+        for (key, value) in members.flatten() {
+            // Try to deserialize the key as a string.
+            // We use `Text` to avoid an heap-allocation.
+            let Some(key_text) = Text::deserialize(&key, "", diagnostics) else {
+                // If this failed, then pass to the next key-value pair.
+                continue;
+            };
+            match key_text.text() {
+                "apiUrl" => {
```

Comment:
I think this is fine, but just flagging that this is a breaking change from previous behavior we inherited from Go with all the weird casings for these keys.

Reply:
It's a change only for the `remoteCache` key in `turbo.json`. All of the other config files still go through the other parser and support all of the variants.